src/FileIO.cpp

```diff
@@ -580,5 +580,24 @@ namespace FileIO {
 
       return (0 == rc);
    }
+
+   Result<std::vector<std::string>> GetDirectoryContents(const std::string& directory) {
```

Comment:
What would happen if there was a sub-directory ? Would that get appended to filesInDirectory as another file?